### PR TITLE
ci: ensures tests are run on push

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

Unit tests job is currently only executed on PRs, leaving pushes on e.g. incubation branch untested. This can lead to issues like the one described in #1235

This PR enables GH Action to be run also for push events.

## How Has This Been Tested?

You can use [`act`](https://github.com/nektos/act) to test GH Actions locally instead of running them on your fork.

```sh
act "on: push" -j unit-test -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:act-latest
```

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
Put a "on hold" label, so we can have some disucssion before merge this change.
Mostly, this involves work process of how PR should be merged, if we enforce all branch up-to-date before PR merge, or how to prevent stable branch gets merged in.

**Comment by zdtsw** [comment]
this is also related to "https://github.com/opendatahub-io/opendatahub-operator/pull/1221#issuecomment-2333328791"

**Comment by ykaliuta** [comment]
> Put a "on hold" label, so we can have some disucssion before merge this change. Mostly, this involves work process of how PR should be merged, if we enforce all branch up-to-date before PR merge, or how to prevent stable branch gets merged in.

Should the workflow just filter by branch?

File: .github/workflows/unit-tests.yaml
Comment (ykaliuta) [inline .github/workflows/unit-tests.yaml:8]: wondering, should it has : here.

Comment (bartoszmajsak) [reply .github/workflows/unit-tests.yaml:8]: It should. Never trust suggestions !

Comment (bartoszmajsak) [reply .github/workflows/unit-tests.yaml:8]:
```suggestion
  pull_request:
```

**Review by ykaliuta — APPROVED** [review]

**Review by zdtsw — APPROVED** [review]

**Pull request merged**
